extensions/jaeger/deployment/src/main/java/io/quarkus/jaeger/deployment/JaegerProcessor.java

```diff
@@ -29,8 +32,11 @@ void setupTracer(JaegerDeploymentRecorder jdr, JaegerBuildTimeConfig buildTimeCo
 
         if (buildTimeConfig.enabled) {
             boolean metricsEnabled = capabilities.isCapabilityPresent(Capabilities.METRICS);
+            MetricsFactory metricsFactory = metricsEnabled
+                    ? new QuarkusJaegerMetricsFactory()
+                    : new NoopMetricsFactory();
 
-            jdr.registerTracer(jaeger, appConfig, metricsEnabled);
+            jdr.registerTracer(jaeger, appConfig, metricsFactory);
```

Comment:
I would keep passing the boolean and initialize the class in the recorder. I think it would work the same and be a bit less confusing.

The recorder method wouldn't be called so the class wouldn't be accessible either.

(Just guessing, you would need to try)

Reply:
But that's exactly how it used to be before, and it was failing :-)

What might work is to have 2 recorder methods, 1 for disabled metrics and 1 for enabled metrics. I guess.

Reply:
I am personally fine with the approach of the PR, IMHO this is what the bytecode recording is all about - being able to easily "pass" objects from build time to runtime.

Reply:
No, it used to be done in `QuarkusJaegerTracer` based on a dynamic value and in a method called at runtime.

I won't fight for it but I prefer when it's clear a runtime object is a runtime object.

Reply:
Ah, I think I see what you mean now. I'll try.

Reply:
Or perhaps not. I have changed `JaegerProcessor` to be

```java
boolean metricsEnabled = capabilities.isCapabilityPresent(Capabilities.METRICS);

jdr.registerTracer(jaeger, appConfig, metricsEnabled);
```

just like before, and then change the `JaegerDeploymentRecorder` to have this:

```java
MetricsFactory metricsFactory = metricsEnabled
        ? new QuarkusJaegerMetricsFactory()
        : new NoopMetricsFactory();

initTracerConfig(jaeger);
quarkusTracer.setMetricsFactory(metricsFactory);
quarkusTracer.reset();
```

But no, this leads to the same failure as before.

I'll try the approach with 2 recorder methods.

Reply:
OK, well, let's leave it at that then. It's not worth your time.

Reply:
The approach with 2 recorder method works:

```java
boolean metricsEnabled = capabilities.isCapabilityPresent(Capabilities.METRICS);

if (metricsEnabled) {
    jdr.registerTracerWithMetrics(jaeger, appConfig);
} else {
    jdr.registerTracer(jaeger, appConfig);
}
```

and

```java
synchronized public void registerTracer(JaegerConfig jaeger, ApplicationConfig appConfig) {
    registerTracer(jaeger, appConfig, new NoopMetricsFactory());
}

synchronized public void registerTracerWithMetrics(JaegerConfig jaeger, ApplicationConfig appConfig) {
    registerTracer(jaeger, appConfig, new QuarkusJaegerMetricsFactory());
}

private void registerTracer(JaegerConfig jaeger, ApplicationConfig appConfig, MetricsFactory metricsFactory) {
    ...
}
```

And I think it's certainly worth my time, as I get to learn a lot!

Reply:
I have modified the PR to use the approach with 2 recorder methods and currently waiting for CI to finish in my fork.